src/bio/bio_monitor.c

```diff
@@ -201,13 +201,20 @@ bio_dev_set_faulty(struct bio_xs_context *xs, enum smd_dev_type st)
 	return rc;
 }
 
+struct get_spdk_cb_arg {
+	struct bio_xs_context	*ctxt;
+	enum smd_dev_type	 st;
+};
+
 static inline struct bio_dev_health *
-xs_ctxt2dev_health(struct bio_xs_context *ctxt)
+cb_arg2dev_health(struct get_spdk_cb_arg *gsca)
 {
-	D_ASSERT(ctxt != NULL);
+
+	struct bio_xs_context	*ctxt = gsca->ctxt;
 	struct bio_xs_blobstore *bxb;
 
-	bxb = bio_xs_context2xs_blobstore(ctxt, SMD_DEV_TYPE_DATA);
+	D_ASSERT(ctxt != NULL);
+	bxb = bio_xs_context2xs_blobstore(ctxt, gsca->st);
```

Comment:
bio_xs_context2bx_blobstore() should be used in this case. And I think we pass the address of bxc_xs_blobstore[st] as completion argument, we could just check if (*arg == NULL) here to tell if bio_xsctxt_free() is underway, right?